inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -191,34 +192,48 @@ def extract_references_from_raw_ref(reference, custom_kbs_file=None):
     )
 
 
-def match_reference(reference):
+def match_reference(reference, previous_matched_recid=None):
```

Comment:
this should be refactored, as it's getting unwieldy. I would suggest splitting this in two functions:
* `match_reference_with_config(reference, config, previous_matched_recid=None)`, that does the actual logic of matching (essentially L213-L214 and L228-L234, properly parameterized) and returns the recid
* `match_reference`, that has a list of configs to use, and calls the previous function with each config in turn until there is a result or the configs are exhausted, and adds the recid to the reference.